### ISSUE TITLE
AMQP.Connection.open/2 issue when passing uri and options

### ISSUE THREAD

**Issue by apellizzn** [pma/amqp#146]
Hi, I am trying to connect to my local rabbitmq but i got the following error
```
 iex(2)> AMQP.Connection.open("amqp://guest:guest@localhost:5672", connection_timeout: 1000)
** (FunctionClauseError) no function clause matching in :amqp_connection.start/2

    The following arguments were given to :amqp_connection.start/2:

        # 1
        {:amqp_params_network, "guest", "guest", "/", 'localhost', 5672, 2047, 0, 10,
         60000, :none,
         [#Function<11.12217282/3 in :amqp_uri.mechanisms/1>,
          #Function<11.12217282/3 in :amqp_uri.mechanisms/1>], [], []}

        # 2
        [connection_timeout: 1000]

    (amqp_client) /Users/apellizz/development/lapin/deps/amqp_client/src/amqp_connection.erl:162: :amqp_connection.start/2
    (amqp) lib/amqp/connection.ex:124: AMQP.Connection.do_open/2
```
By the way if I use `AMQP.Connection.open("amqp://guest:guest@localhost:5672")` everything works fine.

From the documentation https://github.com/pma/amqp/blob/master/lib/amqp/connection.ex it seems calling `open` in the first way should be possible. Any clue on that?
Version: `1.20`

**Comment by ono** [comment]
Hey. Yes, we introduced the option from `1.3.0`. Please try out the new version.

Also note that we have received a bug report on connection URL on `1.3.0`. We are currently investigating it.

https://github.com/pma/amqp/issues/145

**Comment by ono** [comment]
Just released a bug fix with `1.3.1`. Please try out `1.3.1` instead. Thanks!
